tests/suites/test_suite_mpi.function

```diff
@@ -196,6 +211,286 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void mbedtls_mpi_core_io_null()
+{
+    mbedtls_mpi_uint X = 0;
+    int ret;
+
+    ret = mbedtls_mpi_core_read_be( &X, 1, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+    ret = mbedtls_mpi_core_write_be( &X, 1, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+
+    ret = mbedtls_mpi_core_read_be( NULL, 0, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+    ret = mbedtls_mpi_core_write_be( NULL, 0, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+
+    ret = mbedtls_mpi_core_read_le( &X, 1, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+    ret = mbedtls_mpi_core_write_le( &X, 1, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+
+    ret = mbedtls_mpi_core_read_le( NULL, 0, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+    ret = mbedtls_mpi_core_write_le( NULL, 0, NULL, 0 );
+    TEST_ASSERT( ret == 0 );
+
+exit:
+    ;
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void mbedtls_mpi_core_io_be( data_t *input, int nb_int, int nx_64_int, int iret,
+                             int oret )
+{
+    #define BMAX 1024
+    unsigned char buf[BMAX];
+    #define XMAX BMAX / sizeof( mbedtls_mpi_uint )
+    mbedtls_mpi_uint X[XMAX];
+    size_t nx, nb;
+    int ret;
+
+    if( iret != 0 )
+        TEST_ASSERT( oret == 0 );
+
+    TEST_ASSERT( 0 <= nb_int );
+    nb = nb_int;
+    TEST_ASSERT( nb <= BMAX );
```

Comment:
This is where not declaring all the variables at the top works better, since we would only need

```
size_t nb = nb_int;
TEST_ASSERT( nb <= sizeof(buf) );
```

(the assert on `nb` will catch `nb_int < 0`)
(having the variable type adjacent to the variable name means not having to look back to the top of the function to see that this is casting to unsigned)
(I generally prefer using `sizeof(array)` to the value used to size the array - this continues to work even if the array declaration changes in the future)

Reply:
I moved the declarations and removed the macros, but I kept the zero check as it makes it easier both to read and debug and this is a test, the impact on code size is affordable.